ext/sqlite3/database.c

```diff
@@ -259,6 +261,45 @@ busy_handler(int argc, VALUE *argv, VALUE self)
     return self;
 }
 
+static int
+rb_sqlite3_statement_timeout(void *context)
+{
+    sqlite3RubyPtr ctx = (sqlite3RubyPtr)context;
+    struct timespec currentTime;
+    clock_gettime(CLOCK_MONOTONIC, &currentTime);
+
+    if (ctx->stmt_deadline == 0) {
```

Comment:
I don't see it being reset anywhere. I suspect you need to reset the deadline before every query.

Reply:
Great point! The place in `Statement` where I had already access to the database struct was in `prepare`, so I reset it there. Does that make sense?